### ISSUE TITLE
any chance of having 32-bit version for Windows?

### ISSUE THREAD

**Issue by danielkornev** [warrenfalk/rocksdb-sharp#1]
It's fantastic that you've made .NET binding to RocksDB. However, is it possible to make a 32-bit lib and binding to it to run on Windows? We are looking for a 32-bit version for using RocksDB as embedded db in our WPF client app.

Seems like .sh script makes it possible to build 32-bit experimental version only when running under linux

**Comment by warrenfalk** [comment]
I'm glad you might find it useful.  Here is a relevant quote from the Windows port page:

> All building and testing was done for 64-bit. We have not conducted any testing for 32-bit and early reports indicate that it will not run on 32-bit

This is the Microsoft port team speaking who seem to know a lot more about it than I do.

But I can say that Windows isn't even the only one.  At the time I built the script, no 32-bit build of the rocksdb project was possible because of a build error.  I eventually fixed this which uncovered a ton of warnings of integer shortening going on throughout the codebase and when I looked briefly into possibly fixing these, I saw that they seem intractable and possibly serious problems once certain database size thresholds are crossed.  64-bit seems to have been a basic assumption.  I'm guessing this is what Microsoft was talking about.  The rocksdb team did merge my build fix, but simultaneously issued a disclaimer that they can't support 32-bit and so I would be on my own.

So basically 32-bit (barely) builds on Linux/Mac, but you shouldn't rely on it.  Microsoft doesn't think it will work.  It isn't supported, and there is a lot of reason to think that it will eventually corrupt if the data gets large.

And so this is why I didn't devote further time to it, but it might be possible with some modifications to the cmake files, perhaps.

If you are able to get a 32-bit build, I'd be happy to merge it in, but I would recommend that you do whatever you can to make your project 64-bit or at least separate it into 32 and 64 bit components if you do intend to use rocksdb.

**Comment by danielkornev** [comment]
Oh, I see. It's a pity. We'll try to get a 32-bit version of the build, if we'll have any news, we'll let you know. Thank you very much Warren.

We have users on 32-bit machines (or running 32-bit OS, like Windows 7), and we can't leave them w/o our software.

**Comment by danielkornev** [comment]
For now, we've decided to use Redis instead.
